File: python/8kyu/invert_values.py
import codewars_test as test


def invert(lst: list[int]) -> list[int]:
    return [-x for x in lst]


test.it("Basic Tests")
test.assert_equals(invert([1, 2, 3, 4, 5]), [-1, -2, -3, -4, -5])
test.assert_equals(invert([1, -2, 3, -4, 5]), [-1, 2, -3, 4, -5])
test.assert_equals(invert([]), [])

File: typescript/8kyu/invert_values/invert_values.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.invert = void 0;
function invert(array) {
    return array.map(x => -x);
}
exports.invert = invert;
// See https://www.chaijs.com for how to use Chai.
const chai_1 = require("chai");
// TODO Add your tests here
describe("Invert array values", function () {
    it("Basic Tests", function () {
        chai_1.assert.deepEqual(invert([1, 2, 3, 4, 5]), [-1, -2, -3, -4, -5]);
        chai_1.assert.deepEqual(invert([1, -2, 3, -4, 5]), [-1, 2, -3, 4, -5]);
        chai_1.assert.deepEqual(invert([]), []);
        chai_1.assert.deepEqual(invert([0]), [-0]);
    });
});
